jmsSource/src/test/java/io/vantiq/extsrc/jmsSource/TestJMS.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (c) 2019 Vantiq, Inc.
+ *
+ * All rights reserved.
+ * 
+ * SPDX: MIT
+ */
+
+package io.vantiq.extsrc.jmsSource;
+
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeFalse;
+import static org.junit.Assume.assumeTrue;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.jms.JMSException;
+import javax.naming.NamingException;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vantiq.client.SubscriptionCallback;
+import io.vantiq.client.SubscriptionMessage;
+import io.vantiq.client.Vantiq;
+import io.vantiq.client.VantiqResponse;
+
+public class TestJMS extends TestJMSBase {
+    
+    static final int CORE_START_TIMEOUT = 10;
+    
+    static JMSCore core;
+    static JMS jms;
+    static Vantiq vantiq;
+    
+    DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+    String sourceNotificationMessage;
+    
+    StandardOutputCallback sourceNotificationCallback = new StandardOutputCallback();
+    
+    @Before
+    public void setup() {
+        vantiq = new Vantiq(testVantiqServer);
+        vantiq.setAccessToken(testAuthToken);
+        
+        // Create a subscription to source in order to check if message was received as a notification
+        vantiq.subscribe("sources", testSourceName, null, sourceNotificationCallback);
+    }
+    
+    @After
+    public void unsubscribe() {
+        vantiq.unsubscribeAll();
+    }
+        
+    @AfterClass
+    public static void tearDown() {
+        // Delete source from VANTIQ
+        deleteSource();
+        
+        // Close JMSCore if still open
+        if (core != null) {
+            core.close();
+            core = null;
+        }
+        // Close JMS if still open
+        if (jms != null) {
+            jms.close();
+            jms = null;
+        }
+    }
+    
+    @Test
+    public void testProduceAndConsumeQueueMessage() {
+        assumeTrue(jmsDriverLoc != null && testJMSURL != null && testJMSConnectionFactory != null && testJMSInitialContext != null
```

Comment:
since this is used all over, can you move the `assumeTrue` to some method and just call it?  Then easier to change...

Reply:
`checkAllJMSProperties()` or whatever it's called?